File: 0077-combinations/0077-combinations.java
class Solution {
    static List<List<Integer>> answer = new ArrayList<>();
    private void helper(int n,int current,int k,List<Integer> ds){
        if(ds.size() == k){
            answer.add(new ArrayList<>(ds));
            return;
        }
        if(current > n) return;
        for(int i = current+1 ; i <= n ; i++){
            ds.add(i);
            helper(n,i,k,ds);
            ds.remove(ds.size()-1);
        }
       
    }
    public List<List<Integer>> combine(int n, int k) {
        answer.clear();
        helper(n,0,k,new ArrayList<>(1));
        return answer;
    }
}

File: 0077-combinations/0077-combinations.cpp
class Solution {
public:
    vector<vector<int>> result;
    void solve(int k,vector<int> ds,int n,int current){
       if(ds.size() == k){
        result.push_back(ds);
        return;
       }
       for(int i = current ; i <= n ;i++){
        ds.push_back(i);
        solve(k,ds,n,i+1);
        ds.pop_back();
       }
    }
    vector<vector<int>> combine(int n, int k) {
        vector<int> ds;
        solve(k,ds,n,1);
        return result;
    }
};